### DISCUSSION TITLE
The ARC 0.8.3 giving issue on creating listener and "error": "EphemeralRunnerSet.actions.github.com \"eks-mcdevops-runner-j4flc\" is invalid: spec.patchID: Required value"

### DISCUSSION THREAD

**Discussion by aakashvishwakarma** [actions/actions-runner-controller#3393] [Questions]
Hi team, I m deploying and testing ARC  with version 0.8.3 , i find it out that controller got created but when deploying the runnerset neither listener nor runner pod coming up on checking controller logs i find it out that:
Below error it is throwing 
`2024-03-28T16:59:56Z    ERROR   Reconciler error        {"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet", "AutoscalingRunnerSet": {"name":"eks-mcdevops-runner","namespace":"arc-runner"}, "namespace": "arc-runner", "name": "eks-mcdevops-runner", "reconcileID": "c9108a7f-b3e2-4282-9e2c-7d284b002e0f", "error": "EphemeralRunnerSet.actions.github.com \"eks-mcdevops-runner-j4flc\" is invalid: spec.patchID: Required value"}`

On checking the CRDs and go code i get to know that patchID is provided by listener only. So my question is why this behavior all of a sudden in github provided ARC 0.8.3 .

Looking forward to hear back
Thank You
Aakash Vishwakarma

**Comment by aakashvishwakarma** [comment]
Hi @nikola-jokic  may i know if you can help me out debugging the issue and what is the root cause as we are working for enterprise level customer in prod.

Thank you

**Reply by nikola-jokic** [reply]
Hey @aakashvishwakarma,

Did you start upgrading and then saw the error? Can you please write down what did you do before the error started showing up :relaxed: ?

**Reply by aakashvishwakarma** [reply]
Hi @nikola-jokic , We are installing in a 0.8.3 ARC on a  fresh cluster no upgrade.
However the strange thing is that it's only letting us make listener and all pods up and running in 0.9.0 version only.
It is failing in 0.8.3 or even in 0.7.0 as well.

May i know if there is any networking related changes done in 0.9.0 version of ARC so that we can check the networking accordingly in our end.

Thanks

**Reply by nikola-jokic** [reply]
Oh, that probably means that the controller is installed with version 0.9.0. When we build a controller, we set the version. If resources created for that controller are not matching the version, we would delete them right away.
So most likely, the controller version is installed at 0.9.0, while you are installing charts at a version that is different.
Please check the image version of the controller. We tag each one with the version so you can check :relaxed:

**Reply by aakashvishwakarma** [reply]
Sure let me recheck again the controller image tag

**Reply by dcardellino24** [reply]
We managed to fix this error by manually editing the `ephemeralrunnerset` and add a random integer to `spec.patchID`. Then again edit the resource and removed the finalizer. 
Afterwards new resources were created. But thats kinda hacky workaround. Better it should not happen :)

**Reply by rekha-prakash-maersk** [reply]
@dcardellino24 I am running into the same issue while upgrading from 0.8.3 to 0.9.0. Can you please explain how you have added the spec.patchID manually ? I just have the controller running in 0.9.0 version with this error and no listener getting created

**Reply by nocturne1** [reply]
This method allowed me to remove the old `ephemeralrunnerset`:
https://github.com/actions/actions-runner-controller/issues/3395#issuecomment-2030159692

**Reply by rekha-prakash-maersk** [reply]
Hi @nocturne1 , Thank you!! I was able to resolve the issue related to `spec.patchID`. Now the controller still wants to connect to old listener. How can I delete those old scaleset references in the controller , so it will create new listeners from the config ?

**Reply by aakashvishwakarma** [reply]
@dcardellino24  i get it manually  updating is fine but i wanted to know the reason why it happened and is there any  chance it can happen in other new version upgrades?
I m documenting this as we are setting up EKS runners for an Enterprise customers.

**Comment by cheskayang** [comment]
running into the same issue when reverting from 0.9.1 to 0.8.3 due to another error https://github.com/actions/actions-runner-controller/issues/3499
